lib/attrsets.nix

```diff
@@ -339,6 +339,47 @@ rec {
   mapAttrs' = f: set:
     listToAttrs (map (attr: f attr set.${attr}) (attrNames set));
 
+  /* Like mapAttrs, but additionally taking a variable that
+     will be automatically incremented, starting from 0.
+
+     Example:
+       imap0Attrs (i: name: value: value + i)
+          { a = 1; b = 1; c = 1; }
+       => { a = 1; b = 2; c = 3; }
+  */
+  imap0Attrs = f: set:
+    listToAttrs (imap0 (i: attr: nameValuePair attr (f i attr set.${attr})) (attrNames set));
+
+  /* Like imap0Attrs but with the iterator starting at 1
+
+     Example:
+       imap0Attrs (i: name: value: value + i)
+          { a = 1; b = 1; c = 1; }
+       => { a = 2; b = 3; c = 4; }
+  */
+  imap1Attrs = f: set:
+    listToAttrs (imap1 (i: attr: nameValuePair attr (f i attr set.${attr})) (attrNames set));
```

Comment:
I don't think we need that. You can just do ``imap0Attrs (i: name: value: value + i+1)`` instead.

Reply:
we already have imap1 for lists, so it'd be surprising to find we arbitrarily don't have it for attrs

Reply:
I think that is mainly there because it was first. I don't think adding one to an int is such a big inconvenience that we need extra functions for that. It just clutters the lib.

Reply:
Ah, yeah, there's quite a bit of history there. Thanks, I've condensed it into just mapAttrs that start on 0